report/4.Solution/LanuageOfCompiler.tex

```diff
@@ -0,0 +1,37 @@
+\section{The languages of the compiler}
+When creating a compiler there are three languages of relevance, as see in the tombstone diagram\ref{fig:TStoneExample}.
+
+\begin{figure}[H]
+	\centering
+	\includegraphics[width=\textwidth/2+\textwidth/4]{4.Solution/images/T-diagram.png}
+	\caption{
+		Example of Tombstone diagram showcasing an example of three languages for a given compiler.\cite{TStoneWiki}
+	}
+	\label{fig:TStoneExample}
+\end{figure}
+In the case of the HCL compiler the three languages are as follows:
+
+\begin{align*}
+input \to & HCL \\
+output \to & C++/C \\
+compiler \to & Kotlin \\
+\end{align*}
+
+\subsection{Language of the compiler}
+The compiler was written in Kotlin\cite{KotlinWebsite}, a Java based language.
+This was based on a number of factors.
+The group wanted to experiment with using a language they had no prior experience with for the duration of the project, but because of time limitations, a new language had to adhere to already known paradigms.
+The group knew C\# and a bit of Java, however as the HCL language has a lot of functional features, it was deemed important that the group became comfortable with a language that, at the very least was multi paradigm.
+Kotlin is java-based and is similar, in syntax, to C\# and java, which made it easier to learn from the perspective of the group.
+However it is still possible to write methods that is utilizing a functional domain.
+
+This grants the develop af large amount of possibilities, while still being relatively easy to learn.
```

Comment:
This sentence:
Potentially:
"This allows the developer to decide how he/she wants to develop, while still being easy to learn"

Reply:
This sentence was on the first part of it, it sounds strange, Sorry.